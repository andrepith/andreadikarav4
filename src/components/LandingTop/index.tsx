import React from "react";
// import {
//   EnvelopeFill,
//   FileArrowDownFill,
//   ArrowDownCircleFill,
// } from "react-bootstrap-icons";
import bioInterface from "src/lib/Types";

interface LandingTopProps extends bioInterface {
  scrollToExperience: () => void;
}

const LandingTop = ({
  bio: { firstName, nationality, aboutMe, email, resumeLink, social },
  scrollToExperience,
}: LandingTopProps) => {
  return (
    <section className="landing-top position-relative" id="landing-top">
      <div className="container landing-top__text">
        <div className="landing-block">
          <h1 className="landing-title">
            Hey, Iâ€™m {firstName} - A Software <br />
            Engineer from {nationality}.
          </h1>
          <p className="mt-4">{aboutMe}</p>
          <div className="d-flex mt-4 landing-block__item">
            <div className="landing-block__icon">
              <i className="fa fa-envelope-open-text" />
            </div>
            <a className="email-link" href={`mailto:${email}`}>
              {email}
            </a>
          </div>
          <div className="d-flex mt-2 landing-block__item">
            <div className="landing-block__icon">
              <i className="fa fa-file-arrow-down" />
            </div>
            <a
              className="download-link d-block"
              href={resumeLink}
              target="__blank"
            >
              Download Resume
            </a>
          </div>
        </div>
      </div>
      <div className="landing-top__foot" onClick={scrollToExperience}>
        <i className="fa fa-circle-arrow-down fa-beat" />
      </div>
      <ul className="landing-top__side">
        {social.map(({ name, url }, key) => (
          <li key={key}>
            <a href={url} target="__blank">
              <i className={`fa fa-${name.toLocaleLowerCase()}`} />
            </a>
          </li>
        ))}
      </ul>
    </section>
  );
};

export default LandingTop;
