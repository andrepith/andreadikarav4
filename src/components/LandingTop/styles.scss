.landing-top {
  height: 100vh;
  min-height: 500px;
  width: 100%;
  margin: auto;
  display: grid;
  grid-template-areas:
    "a a a"
    "b b d"
    "b b d"
    "c c c";

  &__text {
    grid-area: b;
    align-self: center;
    display: flex;
    justify-content: center;
  }

  &__foot {
    grid-area: c;
    align-self: center;
    justify-self: center;
    border-radius: 50%;
    box-shadow: 0 0 0 0 $color-black;
    cursor: pointer;
    transform: scale(1);
    animation: pulse-black 2s infinite;
  }

  &__side {
    grid-area: d;
    display: flex;
    flex-direction: column-reverse;
    align-items: center;
    margin: 0px;
    padding: 0px;
    list-style: none;
    gap: 20px;

    @include for-desktop-down {
      display: none;
    }

    a {
      color: $color-white;
      i {
        font-size: 34px;
        display: inline-block;
        text-decoration: none;
        text-decoration-skip-ink: auto;
        color: inherit;
        position: relative;
        transition: all 0.25s cubic-bezier(0.645, 0.045, 0.355, 1);
        &:hover {
          transform: translateY(-3px);
          color: $color-highlight;
        }
      }
    }

    &::before,
    &::after {
      content: "";
      display: block;
      width: 1px;
      height: 90px;
      margin: 0px auto;
      background-color: $color-white;
    }
  }
}

@keyframes pulse-black {
  0% {
    transform: scale(0.95);
    box-shadow: 0 0 0 0 rgba(0, 0, 0, 0.7);
  }

  70% {
    transform: scale(1);
    box-shadow: 0 0 0 10px rgba(0, 0, 0, 0);
  }

  100% {
    transform: scale(0.95);
    box-shadow: 0 0 0 0 rgba(0, 0, 0, 0);
  }
}

.landing-title {
  color: $color-white;
  font-size: 3rem;

  @include for-tablet-portrait-down {
    font-size: 30px;
  }
}

.landing-block {
  max-width: 640px;
}

.email-link,
.download-link,
.landing-block p {
  font-weight: 500;
  color: $color-grey;
  font-size: 20px;
}

.email-link,
.download-link,
.landing-block p {
  font-size: 16px;
}

.email-link {
  padding-bottom: 3px;
  border-bottom: 1px solid var(--dark);
  color: $color-white;
  text-decoration: none;
  transition: 0.2s ease-out;
  -webkit-transition: 0.2s ease-out;

  &:hover {
    text-decoration: none;
    background-color: $color-primary;
    color: $color-white;
    border: none;
  }
}

.download-link {
  color: $color-white;
}
