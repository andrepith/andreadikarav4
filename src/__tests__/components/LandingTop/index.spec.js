import renderer from "react-test-renderer";
import LandingTop from "../../../components/LandingTop";

it("Renders correctly", () => {
  const tree = renderer.create(<LandingTop />).toJSON();
  expect(tree).toMatchSnapshot();
});
