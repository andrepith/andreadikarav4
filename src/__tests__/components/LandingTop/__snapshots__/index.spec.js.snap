// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="landing-top position-relative"
  id="landing-top"
>
  <div
    className="container landing-top__text"
  >
    <div
      className="landing-block"
    >
      <h1
        className="landing-title"
      >
        Hey, Iâ€™m 
         - A Software 
        <br />
        Engineer from 
        .
      </h1>
      <p
        className="mt-4"
      />
      <div
        className="d-flex mt-4"
      >
        <div
          className="me-2 mt-1"
        >
          <svg
            fill="white"
            height={16}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M.05 3.555A2 2 0 0 1 2 2h12a2 2 0 0 1 1.95 1.555L8 8.414.05 3.555zM0 4.697v7.104l5.803-3.558L0 4.697zM6.761 8.83l-6.57 4.027A2 2 0 0 0 2 14h12a2 2 0 0 0 1.808-1.144l-6.57-4.027L8 9.586l-1.239-.757zm3.436-.586L16 11.801V4.697l-5.803 3.546z"
            />
          </svg>
        </div>
        <a
          className="email-link"
          href="mailto:undefined"
        />
      </div>
      <div
        className="d-flex mt-2"
      >
        <div
          className="me-2 mt-1"
        >
          <svg
            fill="white"
            height={16}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0H4a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h8a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2zM8 5a.5.5 0 0 1 .5.5v3.793l1.146-1.147a.5.5 0 0 1 .708.708l-2 2a.5.5 0 0 1-.708 0l-2-2a.5.5 0 1 1 .708-.708L7.5 9.293V5.5A.5.5 0 0 1 8 5z"
            />
          </svg>
        </div>
        <a
          className="download-link d-block"
          target="__blank"
        >
          Download Resume
        </a>
      </div>
    </div>
  </div>
  <div
    className="landing-top__foot"
  >
    <svg
      fill="white"
      height={32}
      viewBox="0 0 16 16"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM8.5 4.5a.5.5 0 0 0-1 0v5.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V4.5z"
      />
    </svg>
  </div>
</div>
`;
